### PR TITLE
p2p, rpc, test: address rate-limiting follow-ups

### PR DESCRIPTION
**Pull request by jonatack**
Incorporates review feedback in #22387.

Edit, could be considered separately: should a release note (or two) be added for 22.0? e.g. the new getpeerinfo fields in `Updated RPCs` and the rate-limiting itself in `P2P and network changes`.

### CODE REVIEW

**Comment by theStack** [comment]
Concept ACK

Will code-review tomorrow. At a first glance, most parts of the diff look straight-forward, the change in the rate-limiting logic (swapping the if conditions) needs probably most attention.
To speed up review, adding links to each of the review comments in #22387 that are tackled in this PR could be a good idea.

(the AppVeyor CI build fail seems to be not caused by this PR's changes, by the way)

File: test/functional/p2p_addr_relay.py [hunk 1]
```diff
@@ -257,45 +257,45 @@ def send_addrs_and_test_rate_limiting(self, peer, no_relay, new_addrs, total_add
...
+            self.send_addrs_and_test_rate_limiting(peer, no_relay, new_addrs=200, total_addrs=1410)
 
             # Advance the time by 1000 seconds, permitting the processing of 100 more addresses.
             # Send 200 and verify that 100 are processed.
             self.mocktime += 1000
             self.nodes[0].setmocktime(self.mocktime)
             peer.increment_tokens(100)
 
-            self.send_addrs_and_test_rate_limiting(peer, no_relay, 200, 1610)
+            self.send_addrs_and_test_rate_limiting(peer, no_relay, new_addrs=200, total_addrs=1610)
```
Comment (jonatack) [inline test/functional/p2p_addr_relay.py:295]: review feedback here: https://github.com/bitcoin/bitcoin/pull/22387/files#r671144448

**Review by jonatack — COMMENTED** [review]
> To speed up review, adding links to each of the review comments in #22387 that are tackled in this PR could be a good idea.

Good idea--added!

**Comment by sipa** [comment]
Concept ACK

**Comment by tryphe** [comment]
Concept ACK

Nice fixups! Looks like there's essentially only one functional change here with  `if (rate_limited) {...`.

**Comment by MarcoFalke** [comment]
review ACK b7c4b058ba2625a39666636253620bc08a41c501

Thanks!

Also reset appveyor run.

**Comment by MarcoFalke** [comment]
> Looks like there's essentially only one functional change here with if (rate_limited) {....

I think it still counts as refactoring because the change shouldn't be observable by a user

**Review by theStack — APPROVED** [review]
Code-review ACK b7c4b058ba2625a39666636253620bc08a41c501

**Comment by jonatack** [comment]
Rebased per `git range-diff 2b06af1 b7c4b05 d930c7f`

**Review by theStack — COMMENTED** [review]
re-ACK d930c7f5b091687eb4208a5ffe8a2abe311d8054 🌮 

Checked via `git range-diff b7c4b058...d930c7f5` that the changes since my last ACK were only rebase-related.

**Comment by MarcoFalke** [comment]
review ACK d930c7f5b091687eb4208a5ffe8a2abe311d8054

**Review by Zero-1729 — COMMENTED** [review]
crACK d930c7f

LGTM 🧉

**Pull request merged**